pkgs/development/interpreters/python/mk-python-derivation.nix

```
@@ -309,7 +309,8 @@ let
 
     # Python packages don't have a checkPhase, only an installCheckPhase
     doCheck = false;
-    doInstallCheck = attrs.doCheck or true;
+    # The doCheck argument determines the doInstallCheck attribute.
+    doInstallCheck = doCheck;
```

```suggestion
    doInstallCheck = attrs.doCheck;
```
Could you clarify this with `attrs`?
I think it is confusing since there is a `doCheck = false` right above it.

---

Can't test it right now, but IIRC if we use the `@attrs` pattern, the defaults set with `?` will not be applied when accessing through `attrs.`

---

> Could you clarify this with `attrs`?

`attrs` does not consider the default value given by the set pattern of the build helper. It would be confusing to set a default value in the set pattern (`{ doCheck ? true, ... }`) but shadow it with another default when accessing (`doInstallCheck = attrs.doCheck or true`).

---

Hmmm, indeed, `attrs` seems to be empty, as @TomaSajt pointed out at the beginning.
Then I think `attrs.doCheck or true` would be easier to understand.

The confusion was originally caused by the fact that the argument, `doCheck`, actually evaluates to true even though it defaults to false, right?
Do we need to change this here?

---

> Hmmm, indeed, `attrs` seems empty, as @TomaSajt pointed out at the beginning. Then I think `attrs.doCheck or true` would be easier to understand.

It still shadows the default set by the set pattern, but it sounds like an acceptable trade-off, considering the increased readability.

IMO, the root cause of the readability issue is the choice to specify the `doInstallCheck` attribute with the `doCheck` argument. I'll propose unifying the name (as `doInstallCheck`) in subsequent PR.

---

Addressed.